reward/reward_distributor.go

```diff
@@ -175,13 +177,24 @@ func NewRewardConfig(header *types.Header, rules params.Rules, pset *params.GovP
 	}, nil
 }
 
-func GetTotalTxFee(header *types.Header, rules params.Rules, pset *params.GovParamSet) *big.Int {
+func GetTotalTxFee(header *types.Header, txs []*types.Transaction, receipts []*types.Receipt, rules params.Rules, pset *params.GovParamSet) *big.Int {
 	totalFee := new(big.Int).SetUint64(header.GasUsed)
 	if rules.IsMagma {
 		totalFee = totalFee.Mul(totalFee, header.BaseFee)
 	} else {
 		totalFee = totalFee.Mul(totalFee, new(big.Int).SetUint64(pset.UnitPrice()))
 	}
+
+	if txs == nil || receipts == nil || !rules.IsKaia {
+		return totalFee
+	}
+
```

Comment:
Don't we need to check the length of txs, receipts?
```suggestion
if len(txs) != len(receipts) {
	logger.Error("GetTotalTxFee: txs and receipts length mismatch", "txs", len(txs), "receipts", len(receipts))
	return totalFee
}
```

Reply:
It can protect not to reference receipts when `len(txs) > len(receipts)`. However, this situation should never happen during ApplyTransaction.